runtime/queries/slint/highlights.scm

```diff
@@ -1,122 +1,125 @@
 [
-  (image_macro)
-  (children_macro)
-  (radial_grad_macro)
-  (linear_grad_macro)
-] @function.macro
+  (float_value)
+  (percent_value)
+  (length_value)
+  (physical_length_value)
+  (duration_value)
+  (angle_value)
+  (relative_font_size_value)
+] @constant.numeric.float
 
-(call_expression
-  function: (identifier) @function)
-(call_expression
-  function: (field_expression
-    field: (identifier) @function))
+(simple_identifier) @variable.other
 
-(vis) @keyword.control.import
+(purity) @keyword.storage.modifier
 
-(transition_statement state: (identifier) @variable.other.member)
-(state_expression state: (identifier) @variable.other.member)
-(struct_block_definition field: (identifier) @variable.other.member)
-(assign_property (identifier) @attribute)
+(function_visibility) @keyword.storage.modifier
 
-(comment) @comment
+(property_visibility) @keyword.storage.modifier
 
-(string_literal) @string
-(int_literal) @constant.numeric.integer
-(float_literal) @constant.numeric.float
+(animate_option_identifier) @keyword
 
-[
-  "in"
-  "in-out"
-  "for"
-] @keyword.control.repeat
+(builtin_type_identifier) @type.builtin
 
-[
-  "import"
-  "export"
-  "from"
-] @keyword.control.import
+(reference_identifier) @variable.builtin
 
-[
-  "if"
-  "else"
-  "when"
-] @keyword.control.conditional
+(type
+  [
+    (type_list)
+    (user_type_identifier)
+    (anon_struct_block)
+  ]) @type
 
-[
-  "struct"
-  "property"
-] @keyword.storage.type
+(user_type_identifier) @type
 
 [
-  "global"
-] @keyword.storage.modifier
+  (comparison_operator)
+  (mult_prec_operator)
+  (add_prec_operator)
+  (unary_prec_operator)
+  (assignment_prec_operator)
+] @operator
 
+; Functions and callbacks
+(argument) @variable.parameter
 
-[
-  "root"
-  "parent"
-  "duration"
-  "easing"
-] @variable.builtin
+(function_call) @function
 
+; definitions
+(callback
+  name: (_) @function)
 
-[
-  "callback"
-  "animate"
-  "states"
-  "out"
-  "transitions"
-  "component"
-  "inherits"
-] @keyword
+(component
+  id: (_) @variable)
 
-[
-  "black"
-  "transparent"
-  "blue"
-  "ease"
-  "ease_in"
-  "ease-in"
-  "ease_in_out"
-  "ease-in-out"
-  "ease_out"
-  "ease-out"
-  "end"
-  "green"
-  "red"
-  "start"
-  "yellow"
-  "white"
-  "gray"
- ] @constant.builtin
+(enum_definition
+  name: (_) @type.enum)
 
-[
-  "true"
-  "false"
-] @constant.builtin.boolean
+(function_definition
+  name: (_) @function)
+
+(property
+  name: (_) @variable.other.member)
+
+(struct_definition
+  name: (_) @type)
+
+(typed_identifier
+  name: (_) @variable)
+
+(typed_identifier
+  type: (_) @type)
+
+(binary_expression
+  op: (_) @operator)
 
-"@" @keyword
+":=" @operator
 
-; ; Punctuation
+(unary_expression
+  op: (_) @operator)
+
+(if_statement
+  "if" @keyword.conditional)
+
+(if_statement
+  ":" @punctuation.delimiter)
+
+(if_expr
+  [
+    "if"
+    "else"
+  ] @keyword.conditional)
+
+(ternary_expression
+  [
+    "?"
+    ":"
+  ] @keyword.conditional)
```

Comment:
```suggestion
  "if" @keyword.control.conditional)

(if_statement
  ":" @punctuation.delimiter)

(if_expr
  [
    "if"
    "else"
  ] @keyword.control.conditional)

(ternary_expression
  [
    "?"
    ":"
  ] @keyword.control.conditional)
```

(to align with https://docs.helix-editor.com/master/themes.html#syntax-highlighting)